### ISSUE TITLE
DOC: Add documentation for each task

### ISSUE THREAD

**Issue by scott-huberty** [scott-huberty/q1k-neuroimaging#4]
@gabe33033 @dsrishyla I'd like us to move the task documentation from the [SOP on Sharepoint](https://mcgill.sharepoint.com/:w:/r/sites/Q1K_Group/_layouts/15/Doc.aspx?sourcedoc=%7BC545167A-CF54-467D-A42C-FD3F9A5D9526%7D&file=7.%20SOP%20EEG-ET.docx&action=default&mobileredirect=true). Please choose 2 or 3 tasks to do this for.  @dsrishla probably PLR for you plus whatever other task you are interested in working on.

As an example of how to do this, I started with Gap-Overlap. See [the new web page](https://q1k-neuroimaging.readthedocs.io/en/latest/), the current `docs/source/` directory structure, and #3 for an example of how I added a Gap Overlap Page to the docs. 

A couple of key things in that PR

- I added additional sphinx extension to `requirements.txt`. Run `pip install -r requirements.txt` to with your virtual env active, to make sure your environment has the extra dependencies to build the new doc.
- There should be a separate `.rst` file for each task in `docs/source/tasks/`. Then add the task to `docs/source/index.rst` the way I did for gap overlap.

A couple of other hints.

When writing new documentation RsT files (using your VS code editor!). You can test documentation building locally.

```
# assuming you start from the q1k-neuroimaging directory
cd docs
# build the doc
make html
# check for warnings in your terminal output, then open the built doc
open build/html/index.rst
# when you are done, remove the built docs to keep docs/build clean.
make clean
```

Always make sure that you create a new branch before making your changes. typically I do

```
git checkout main
# upstream should be the git url to scott-huberty/q1k-neuroimaging
git pull upstream main 
# create new branch
git checkout -b new_feature_branch_name
```

LMK if you have any questions.

- [x] Rest #16 
- [x] Antisaccade #19 
- [x] SSVEP #14 
- [x] Visual Search #19 
- [x] SSAEP #15 
- [x] Gap-overlap #5 
- [x] MMN #17 
- [ ] PLR
- [x] NSP #6
- [x] FSP #19

**Comment by scott-huberty** [comment]
@dsrishyla and @gabrielblancogomez, see the event code table I added to the [Gap Overlap Page](https://q1k-neuroimaging.readthedocs.io/en/latest/tasks/gap_overlap.html#go). This was lifted directly from our [sharepoint documentation,](https://mcgill.sharepoint.com/:w:/r/sites/Q1K_Group/_layouts/15/Doc.aspx?sourcedoc=%7B6BF97E21-91C6-4BDD-A111-89358EAB0950%7D&file=2_Task-SOP_EEG-ET.docx&action=default&mobileredirect=true) but I revised it a bit to be very clear about the links between the Eyelink codes and the EGI codes. We will also have to update the DIN codes associated with each task, as those are for sure outdated. But in general we should start by getting this table on the board for each task, and then correcting mistakes while we are analyzing specific tasks.

I think/hope that this will really help to synergize the documentation with the analyses, i.e. make it easier for you to reference the documentation when you do your analyses, and make fixes as needed via PR's.
